realm/realm-jni/src/io_realm_internal_tableview.cpp

```diff
@@ -165,9 +170,11 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableView_nativeGetSourceRowIndex
 {
     try {
         if (!VIEW_VALID_AND_IN_SYNC(env, nativeViewPtr))
-            return npos;
+            return -1;
         if (!ROW_INDEX_VALID(env, TV(nativeViewPtr), rowIndex))
-            return npos;
+            return -1;
+        if (!TV(nativeViewPtr)->is_row_attached(rowIndex))
+            return -1;
     } CATCH_STD()
     return TV(nativeViewPtr)->get_source_ndx(S(rowIndex));   // noexcept
```

Comment:
I think it was @beeender who explained to me why `*_VALID` should be called outside `try/catch`. When `CATCH_STD()` catches a throw via `ThrowException()` from `*_VALID`, it _again_ calls `ThrowException` to throw the same exception. So, there is no need to call `*_VALID` from `try/catch`.